### PR TITLE
TINY-8102: Replace badly named APIs

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: TINY-8102

Description of Changes:
* Renamed the`tinymce.html.schema.getWhiteSpaceElements()` API to `tinymce.html.schema.getWhitespaceElements()`
* Deprecated the `fire()` function and replaced it with the `dispatch()` function in the following places: [tinymce.util.EventDispatcher](https://www.tiny.cloud/docs/api/tinymce.util/tinymce.util.eventdispatcher/), [tinymce.util.Observable](https://www.tiny.cloud/docs/api/tinymce.util/tinymce.util.observable/), [tinymce.dom.DOMUtils](https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/), [tinymce.dom.EventUtils](https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.eventutils) and [tinymce.Editor](https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/)

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts [hunk 1]
```diff
@@ -126,16 +126,16 @@ class EventDispatcher<T> {
...
+   * @method dispatch
+   * @param {String} name Name of the event to dispatch
    * @param {Object?} args Event arguments.
    * @return {Object} Event args instance passed in.
    * @example
-   * instance.fire('event', {...});
+   * instance.dispatch('event', {...});
    */
-  public fire <K extends string, U extends MappedEvent<T, K>>(name: K, args?: U): EditorEvent<U> {
+  public dispatch <K extends string, U extends MappedEvent<T, K>>(name: K, args?: U): EditorEvent<U> {
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts:138]: Are we sure we don't want to keep a deprecated fire function around this seems like a pretty massive change to just rename it. Lots of our internal plugins will break and so will likely third party code as well if they use our event system. I'm all for us changing this but we have not had any deprecation messages out for this have we. We don't know how much it's used in the wild etc?

- [ ] Consider adding deprecated fire alternative functions

Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts:138]: Yeah this was Andy and I mentioned late yesterday afternoon as well. Ben is going to add the `fire` function back and mark it as deprecated. It looks like he moved the Jira back out of review but didn't update this PR (or convert it back to a draft). I'll go ahead and do that now.

**Comment by lnewson** [comment]
Converting back to a draft for now while some additional changes are made (see the previous comments).

File: modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts [hunk 2]
```diff
@@ -1804,15 +1806,28 @@ const DOMUtils = (doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
...
     /**
-     * Fires the specified event name with object on target.
-     *
+     * Fires the specified event on the specified target.
+     * <br>
+     * <em>Deprecated in TinyMCE 6.0 and has been marked for removal in TinyMCE 7.0. Use the <code>dispatch<code> instead.</em>
      * @method fire
-     * @param {Node/Document/Window} target Target element or object to fire event on.
-     * @param {String} name Name of the event to fire.
+     * @param {Object} target Target node/window or custom object.
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts:1813]:
```suggestion
     * @param {Node/Document/Window} target Target node/window or custom object.
```

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -81,6 +81,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The Editor query command APIs will now return `false` or an empty string on removed editors #TINY-7829
 - The `plugins` option now returns a `string` array instead of a space separated string #TINY-8455
 - Replaced 'Powered by Tiny' link text with logo #TINY-8371
+- The `fire()` function of `tinymce.Editor`, `tinymce.dom.EventUtils`, `tinymce.dom.DOMUtils`, `tinymce.util.Observable` and `tinymce.util.EventDispatcher` is now deprecated. Use `dispatch()` function instead #TINY-8102
```
Comment (ltrouton) [inline modules/tinymce/CHANGELOG.md:84]: Do we need a changelog entry for the `schema.getWhitespaceElements` change?

File: modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts [hunk 4]
```diff
@@ -1804,15 +1806,28 @@ const DOMUtils = (doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
...
+     * @param {String} name Event name to fire.
+     * @param {Object} args Optional arguments to send to the observers.
+     * @return {EventUtils} Event utils instance.
+     * @deprecated Use dispatch() instead
+     */
+    fire,
+
+    /**
+     * Dispatches the specified event name with object on target.
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts:1822]: I am not sure on the exact wording but maybe something like this?
```suggestion
     * Dispatches the specified event name and optional object on the specified target.
```

**Review by ltrouton — APPROVED** [review]
They might be there but I have missed it. Are there still some tests to make sure that the `fire` still works as expected?

**Comment by tiny-ben-tran** [comment]
> They might be there but I have missed it. Are there still some tests to make sure that the `fire` still works as expected?

All are replaced with `dispatch()`. I was tempting to leave some but not sure what to do for deprecated stuffs

File: modules/tinymce/src/core/main/ts/api/html/Schema.ts [hunk 5]
```diff
@@ -940,10 +940,10 @@ const Schema = (settings?: SchemaSettings): Schema => {
   /**
    * Returns a map with elements where white space is to be preserved. For example: <code>&#60;pre&#62;</code> or <code>&#60;script&#62;</code>.
    *
-   * @method getWhiteSpaceElements
+   * @method getWhitespaceElements
    * @return {Object} Name/value lookup map for white space elements.
    */
-  const getWhiteSpaceElements = Fun.constant(whiteSpaceElementsMap);
+  const getWhitespaceElements = Fun.constant(whiteSpaceElementsMap);
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/html/Schema.ts:946]: Can you also please rename the `whiteSpaceElementsMap` to match this name please?

File: modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts [hunk 6]
```diff
@@ -127,15 +127,31 @@ class EventDispatcher<T> {
 
   /**
    * Fires the specified event by name.
-   *
+   * <br>
+   * <em>Deprecated in TinyMCE 6.0 and has been marked for removal in TinyMCE 7.0. Use the <code>dispatch<code> API instead.</em>
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts:131]:
```suggestion
   * <em>Deprecated in TinyMCE 6.0 and has been marked for removal in TinyMCE 7.0. Use <code>dispatch<code> instead.</em>
   *
```

File: modules/tinymce/src/core/main/ts/api/util/Observable.ts [hunk 7]
```diff
@@ -41,30 +42,49 @@ const Observable: Observable<any> = {
   /**
    * Fires the specified event by name. Consult the
    * <a href="/docs/advanced/events">event reference</a> for more details on each event.
+   * <br>
+   * <em>Deprecated in TinyMCE 6.0 and has been marked for removal in TinyMCE 7.0. Use the <code>dispatch<code> instead.</em>
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/util/Observable.ts:46]:
```suggestion
   * <em>Deprecated in TinyMCE 6.0 and has been marked for removal in TinyMCE 7.0. Use <code>dispatch<code> instead.</em>
```

File: modules/tinymce/src/core/main/ts/html/ParserFilters.ts [hunk 8]
```diff
@@ -59,7 +59,7 @@ const register = (parser: DomParser, settings: DomParserSettings): void => {
     parser.addNodeFilter('br', (nodes, _, args) => {
       const blockElements = Tools.extend({}, schema.getBlockElements());
       const nonEmptyElements = schema.getNonEmptyElements();
-      const whiteSpaceElements = schema.getWhiteSpaceElements();
+      const whiteSpaceElements = schema.getWhitespaceElements();
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/html/ParserFilters.ts:62]: Can you also please rename this variable to match?

File: modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts [hunk 9]
```diff
@@ -256,7 +256,7 @@ describe('browser.tinymce.core.html.SchemaTest', () => {
 
   it('getWhiteSpaceElements', () => {
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts:257]:
```suggestion
  it('getWhitespaceElements', () => {
```

**Review by lnewson — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 10]
```diff
@@ -81,6 +81,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The Editor query command APIs will now return `false` or an empty string on removed editors #TINY-7829
 - The `plugins` option now returns a `string` array instead of a space separated string #TINY-8455
 - Replaced 'Powered by Tiny' link text with logo #TINY-8371
+- Replaced the `fire()` function with a `dispatch()` in various APIs #TINY-8102
+- Renamed the `tinymce.html.schema.getWhiteSpaceElement()` API to `tinymce.html.schema.getWhitespaceElements()` #TINY-8102
```
Comment (TheSpyder) [inline modules/tinymce/CHANGELOG.md:85]:
```suggestion
- Renamed the `tinymce.html.schema.getWhiteSpaceElements()` API to `tinymce.html.schema.getWhitespaceElements()` #TINY-8102
```

That typo in the old name made me think this was going to break RTC :P

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 11]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New `imagepreview` dialog component, allowing preview and zoom of any image URL #TINY-8333
 - New `editor.annotator.removeAll` API to remove all annotations by name #TINY-8195
 - New `Resource.unload` API to make it possible to unload resources #TINY-8431
+- New `dispatch()` function has been added to replace the old `fire()` function in various APIs #TINY-8102
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:20]:
```suggestion
- New `dispatch()` function to replace the now deprecated `fire()` function in various APIs #TINY-8102
```

File: modules/tinymce/CHANGELOG.md [hunk 12]
```diff
@@ -81,6 +82,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The Editor query command APIs will now return `false` or an empty string on removed editors #TINY-7829
 - The `plugins` option now returns a `string` array instead of a space separated string #TINY-8455
 - Replaced 'Powered by Tiny' link text with logo #TINY-8371
+- Renamed the `tinymce.html.schema.getWhiteSpaceElements()` API to `tinymce.html.schema.getWhitespaceElements()` #TINY-8102
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:85]:
```suggestion
- Renamed the `getWhiteSpaceElements()` function to `getWhitespaceElements()` in the `Schema` API #TINY-8102
```

**Pull request merged**